gsy_framework/matching_algorithms/preferred_partners_algorithm.py

```diff
@@ -97,15 +97,15 @@ def _get_actor_to_offers_mapping(
         """Map seller ids/origin ids to their offers list."""
         mapping = {}
         for offer in offers:
-            if offer["seller_id"]:
-                if offer["seller_id"] not in mapping:
-                    mapping[offer["seller_id"]] = []
-                mapping[offer["seller_id"]].append(offer)
-            if (offer["seller_origin_id"]
-                    and offer["seller_origin_id"] != offer["seller_id"]):
-                if offer["seller_origin_id"] not in mapping:
-                    mapping[offer["seller_origin_id"]] = []
-                mapping[offer["seller_origin_id"]].append(offer)
+            if offer["seller"]["uuid"]:
+                if offer["seller"]["uuid"] not in mapping:
+                    mapping[offer["seller"]["uuid"]] = []
+                mapping[offer["seller"]["uuid"]].append(offer)
+            if (offer["seller"]["origin_uuid"]
+                    and offer["seller"]["origin_uuid"] != offer["seller"]["uuid"]):
```

Comment:
I actually tried to match the behavior of the former implementation and I assumed that it will always be present (in theory it will) however I agree with you, especially WRT future extensibility of the code. I'll treat these fields as optional in order to protect ourselves from cases in the future where the seller is not transmitted at all. Thanks!

Reply:
Fixed